spec/unit/type/component_spec.rb

```diff
@@ -43,7 +43,7 @@
     end
 
     it "should produce the class name even for the class named main" do
-      expect(component.new(:name => "Class[main]").pathbuilder).to eq(["Main"])
+      expect(component.new(:name => "Class[main]").pathbuilder).to eq(['main'])
```

Comment:
If `main` is a reserved class name, then the above test should be removed. Instead, we should trap all attempts to create a class with that name. The only time a Class[main] is created is when the [compiler executes `#evaluate_main`](/puppetlabs/puppet/blob/master/lib/puppet/parser/compiler.rb#L580).